### DISCUSSION TITLE
Help to solve not "valid dictionary or object" error.

### DISCUSSION THREAD

**Discussion by gety9** [fastapi/fastapi#12390] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class Message(BaseModel):
    wtchtag: str
    wtchnm: str
    wtchurl: str
    diff: str
    diffurl: str

class Diff(BaseModel):
    version: str
    title: str
    message: Message

@app.post("/push_diff/")
async def push_diff(diff_data: Diff):
    # logic here
```

### Description

I am trying to make fastapi receive json notifications from changedetection_io, but i am getting "Input should be a valid dictionary or object to extract fields from" on message field. (looking at message field error seems to be caused by escapes)

If i send same json load (as changedetection_io is sending (except i replace `'` with `"`)) using fastapi's docs it works (200 response). But how to make it work when changedetection_io is sending it?

Screenshots: 
fastapi's docs "try it out" - works, response 200
![Screenshot_2024-10-05_17-33-15](https://github.com/user-attachments/assets/fdd15358-c6ce-4b64-b714-0dbbfdd65be7)

changedetection_io log (contains json payload and error)
![Screenshot_2024-10-05_17-27-45](https://github.com/user-attachments/assets/622e5d79-68c3-4b73-bc14-d9adf8abff09)

### Operating System

Linux, Other

### Operating System Details

android 11, termux (linux like envinoment on android)

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

3.11.10

### Additional Context

I checked "I already checked if it is not related to FastAPI but to Pydantic." but honestly i am not sure. I searched both FastAPI's and Pydantic's github issues for something similar and nowhere i see answers. Even stockoverflow have only 2 questions for search "Input should be a valid dictionary or object to extract fields from", but they didn't help with my situation.

**Comment by JavierSanchezCastro** [comment]
It may be because the POST call does not send the Content-Type header with "application/json" as value.

**Reply by gety9** [reply]
@JavierSanchezCastro thank you for suggestion.
It didn't help. But it did change something (more escapes, and message field is in input now (nested))

Screenshot of log: 
(reversed order)
(test 1 without Content-Type header
test 2 with Content-Type header = application/json (did nothing)
test 3 Content-Type header = "application/json" (nested, more escapes))
![Screenshot_2024-10-06_17-54-45](https://github.com/user-attachments/assets/4b11aea4-bcdd-49df-9f00-80fe61cec5f0)

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
